kratos/spatial_containers/search_structure.h

```diff
@@ -394,36 +398,45 @@ class SearchStructure
             Block *= MaxSize_[i-1];
             Axis[i].Set(Min_[i],Max_[i],MaxSize_[i],Block);
         }
+
+        hasIterators = false;
     }
 
     IndexType BeginRow(IndexType const& Idx)
     {
         return Idx + Axis[0].Min;
     }
+
     IndexType EndRow(IndexType const& Idx)
     {
         return Idx + Axis[0].Max+1;
     }
 
-
     SearchStructure const& operator++()
     {
         for(SizeType i = 0; i < Dimension; i++)
             ++(Axis[i]);
         
-        RowBegin = DataBegin + Axis[0].Min;
-        RowEnd   = DataBegin + Axis[0].Max + 1;
+        // Update the row iterators only if the DataBegin iterator exists
+        if(hasIterators) {
```

Comment:
doesn't this imply a performance penalty for everything?

Reply:
Well yes, but only for the ++ or -- operators of the search structure, which are not widely used. I can make some peformance tests if you feel more confortable but it should not have a real impact on the performance...